WordPressKit/Domains/DomainsServiceRemote+AllDomains.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+
+extension DomainsServiceRemote {
+
+    // MARK: - API
+
+    public func fetchAllDomains(params: AllDomainsEndpointParams? = nil, completion: @escaping (AllDomainsEndpointResult) -> Void) {
+        let path = self.allDomainsEndpointPath(params: params)
+        self.wordPressComRestApi.GET(path, parameters: nil) { result, _ in
+            do {
+                switch result {
+                case .success(let result):
+                    let data = try JSONSerialization.data(withJSONObject: result, options: [])
+                    let decoder = JSONDecoder()
+                    decoder.dateDecodingStrategy = .iso8601
+                    let result = try decoder.decode(AllDomainsEndpointResponse.self, from: data)
+                    completion(.success(result.domains))
+                case .failure(let error):
+                    throw error
+                }
+            } catch let error {
+                completion(.failure(error))
+            }
+        }
+    }
+
+    private func allDomainsEndpointPath(params: AllDomainsEndpointParams?) -> String {
+        let endpoint = "all-domains"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._1_1)
+        var components = URLComponents(string: path)
+        var queryItems = [URLQueryItem]()
+        if let noWPCOM = params?.noWPCOM {
+            queryItems.append(.init(name: "no_wpcom", value: "\(noWPCOM)"))
+        }
+        if let resolveStatus = params?.resolveStatus {
+            queryItems.append(.init(name: "resolve_status", value: "\(resolveStatus)"))
+        }
+        if let locale = params?.locale {
+            queryItems.append(.init(name: "locale", value: locale))
+        }
+        components?.queryItems = queryItems
+        return components?.url?.absoluteString ?? path
+    }
+
+    // MARK: - Public Types
+
+    public typealias AllDomainsEndpointResult = Result<[AllDomainsListItem], Error>
+
+    public struct AllDomainsEndpointParams {
+        public var resolveStatus: Bool?
+        public var locale: String?
+        public var noWPCOM: Bool?
+        public init() {}
+    }
```

Comment:
Resolved in 1703cfb8bdf526b420302e814e3276d0164f2d62

Reply:
Looking at the backend code, it seems `no_wpcom` is always valid on the endpoint, as is `resolve_status`. Could these use good defaults to avoid nullability?

Reply:
@jkmassel What do you mean by they're valid on the endpoint?

Reply:
@salimbraksa I think what Jeremy means here is that the endpoint will always be able to handle these parameters. So if we always send a value (the same value as the default values on the endpoint) that would save us the hassle of handling optionals.
The default values on the endpoint are:
* `no_wpcom`: false
* `resolve_status`: false
Ref: fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Sjcpbz%2Qwfba%2Qraqcbvagf%2Spynff.jcpbz%2Qfgber%2Qqbznvaf%2Qncv%2Qraqcbvagf.cuc%3Se%3Q28672n42%23927-og

Reply:
Thanks @hassaanelgarem. Resolved in 423a5f005aeb1bb82d32c69504fff9fe8c396d81.